src/Generating/Trees.cpp

```diff
@@ -29,6 +29,37 @@ static const sCoords Corners[] =
 	{1, 1},
 } ;
 
+// Array with possible directions for a branch to go to.
+static const Vector3d & pickBranchDirection(Vector3i a_BlockPos, int a_Seq)
+{
+	static const std::array<Vector3d, 32> directions =
+	{
+		{
+			{ -1, 0, 0 }, { 0, 0, -1  },
+			{ -1, 0, 1 }, { -1, 0, -1 },
+			{ 1, 0, 1  }, { 1, 0, -1  },
+			{ 1, 0, 0  }, { 0, 0, 1   },
+
+			{ -0.5, 0, 0   }, { 0, 0, -0.5    },
+			{ -0.5, 0, 0.5 }, { -0.5, 0, -0.5 },
+			{ 0.5, 0, 0.5  }, { 0.5, 0, -0.5  },
+			{ 0.5, 0, 0    }, { 0, 0, 0.5     },
+
+			{ -1, 0.5, 0 }, { 0, 0.5, -1  },
+			{ -1, 0.5, 1 }, { -1, 0.5, -1 },
+			{ 1, 0.5, 1  }, { 1, 0.5, -1  },
+			{ 1, 0.5, 0  }, { 0, 0.5, 1   },
+
+			{ -0.5, 0.5, 0   },  { 0, 0.5, -0.5    },
+			{ -0.5, 0.5, 0.5 },  { -0.5, 0.5, -0.5 },
+			{ 0.5, 0.5, 0.5  },  { 0.5, 0.5, -0.5  },
+			{ 0.5, 0.5, 0    },  { 0, 0.5, 0.5     },
+		}
+	};
+
+	return directions[(static_cast<size_t>(a_BlockPos.SqrLength()) + a_Seq) % directions.size()];
```

Comment:
I don't think `SqrLength` is a good replacement, you should still use `a_Noise.IntNoise3DInt()`